jax/experimental/mesh_utils.py

```diff
@@ -25,6 +25,35 @@
 _TPU_V3 = 'TPU v3'
 _TPU_V4 = 'TPU v4'
 
+# Maps physical topology -> mesh shape -> transpose to use for jekbradbury's
+# famous contiguous mesh trick. The trick only works for certain topologies and
+# mesh shapes, which are chosen to be representative of large language model
+# workloads (the three axes are [replica, data, model]).
+_TRANSPOSE_TRICKS = {
+    (2, 2, 1): {
+        (1, 2, 2): (0, 1, 2),
+        (2, 2): (0, 1, 2),
+    },
+    (2, 2, 4): {
+        (1, 4, 4): (0, 1, 2),
+        (4, 4): (0, 1, 2),
+    },
+    (4, 4, 4): {
+        (1, 16, 4): (0, 2, 1),
+        (16, 4): (0, 2, 1),
+    },
+    (4, 8, 8): {
+        (1, 64, 4): (0, 2, 1),
+        (1, 4, 64): (0, 2, 1),
+        (64, 4): (0, 2, 1),
+        (4, 64): (0, 2, 1),
+    },
+    (8, 8, 8): {
```

Comment:
Missing (8, 16, 16), which is a valid and useful slice size where the trick also works

Reply:
Nice, added that one.